src/common/commonutils/UrlUtils.c

```diff
@@ -3,24 +3,26 @@
 
 #include "Internal.h"
 
-char* UrlEncode(char* target)
+char* UrlEncode(const char* target, size_t targetSize)

```

Comment:
Is there a possibility that we pass a non-terminated string here? I'm not sure why we add the size here. If that's really need then OK.

Reply:
The answer in actual use is no. Could fuzzing call this with non null terminated strings?

If you look where these functions are used, only Encode is used in a single place, on a statically allocated and null terminated buffer buffer. Decode is just used in the unit test to validate the Encode one. These are very limited functionality functions which are not used in ASB.

Reply:
I agree the size looks bad, would only be truly useful if we had SAL annotations here but we don't.